### PR TITLE
Adding CHANGELOG auto generation script

### LINKED ISSUE #76: Auto-generate the CHANGELOG.md file upon release
#### Issue Summary

For every release, we currently hand craft a CHANGELOG.md update, now we would like to automate this process. Please see the existing CHANGELOG.md for formatting structure.

#### Acceptance Criteria

* A script, which when run, creates a CHANGELOG.md update based on any merged PRs since the last release.

### PR DESCRIPTION
**Pull request by crnbrdrck**
# Fixes #76  

### Checklist
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guide] and my PR follows them.
- [x] I updated my branch with the master branch.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation about the functionality in the appropriate .md file
- [x] I have added in line documentation to the code I modified

### Short description of what this PR does:
- Adds a script `changes.py` which, when run after creating a new tag in the repo, will parse commit messages for merged PRs and automatically update the `CHANGELOG.md` with the information
- See [here](https://github.com/crnbrdrck/smtpapi-python/blob/changelog-example/CHANGELOG.md) for an example of the CHANGELOG having been modified by this script after I created a `v0.4.0` tag from the current `HEAD` of master

If you have questions, please send an email to [Sendgrid](mailto:dx@sendgrid.com), or file a Github Issue in this repository.

### CODE REVIEW

**Comment by SendGridDX** [comment]
[CLA assistant check](https://cla.sendgrid.com/sendgrid/smtpapi-python?pullRequest=80) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.sendgrid.com/sendgrid/smtpapi-python?pullRequest=80) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.sendgrid.com/check/sendgrid/smtpapi-python?pullRequest=80) it.</sub>

File: changes.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
+"""
```
Comment (misterdorm) [inline changes.py]: Let's do `#!/usr/bin/python` here so this script can easily be run directly from a shell.

File: changes.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    # Create and write out the new version information
+    latest_release_date_string = latest_release_date.strftime('%Y-%m-%d')
+    f.write('## [{}] - {} ##\n'.format(
+        latest_release,
+        latest_release_date_string,
+    ))
+    # Add the stuff that was added
+    f.write('### Added\n')
+    for commit in added:
+        f.write('- {}: {}{} Big thanks to {} for the PR!\n'.format(
```
Comment (misterdorm) [inline changes.py]: I'd prefer a shorter `(via {})` string here, just to keep the changelog less noisy.

File: changes.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+            generate_pr_link(commit['number']),
+            commit['title'],
+            '.' if commit['title'][-1] != '.' else '',
+            generate_user_link(commit['user'])
+        ))
+    f.write('\n')
+    # Add the fixes
+    f.write('### Fixes\n')
+    for commit in fixes:
+        f.write('- {}: {}{} Big thanks to {} for the PR!\n'.format(
```
Comment (misterdorm) [inline changes.py]: And here, too.

File: changes.py (hunk 2 above)
Comment (crnbrdrck) [reply changes.py]: Sure thing! Wasn't really sure what pattern to use so I took the one from the other repos :sweat_smile:

**Comment by thinkingserious** [comment]
Hello @crnbrdrck,<br /><br/>Thanks again for the PR!<br />                                              <br />It's <a href="https://sendgrid.com/blog/hacktoberfest-2018-has-arrived/">HACKTOBERFEST</a>! We want to show our appreciation by sending                                              you some special Hacktoberfest swag. If you have not already, could you please fill out                                              <a href="https://dx.sendgrid.com/hacktoberfest/join">this form</a> so we can send it to you? Thanks!                                             <br /><br />Team SendGrid DX

**Comment by thinkingserious** [comment]
Hello @crnbrdrck,<br /><br/>Thanks again for the PR!<br /><br />                                         You have earned 7 out of the 5 points needed to receive glorious SendGrid Hacktoberfest swag.<br /><br />                                         Please take a moment to checkout <a href="https://dx.sendgrid.com/hacktoberfest/issues">this link</a> to find more issues to get you past the required threshold or to simply continue the celebration.<br /><br />                                         Also, <strong>please be sure you have officially registered with us <a href="https://dx.sendgrid.com/hacktoberfest/join">here</a> by November 1, 2018</strong> to qualify.<br /><br />                                         If you have any questions you can email us at <a href="mailto:dx+hacktoberfest2018@sendgrid.com">dx+hacktoberfest2018@sendgrid.com</a>.<br /><br />                                         Thank you and Happy Hacktobering!<br /><br />                                         Team SendGrid

**Pull request merged**